public/app/features/scenes/variables/variants/query/QueryVariable.tsx

```diff
@@ -0,0 +1,162 @@
+      query: '',
+      options: [],
+      datasource: null,
+      regex: '',
+      refresh: VariableRefresh.onDashboardLoad,
+      sort: VariableSort.alphabeticalAsc,
+      ...initialState,
+    });
+  }
+
+  public activate(): void {
+    super.activate();
+    const timeRange = sceneGraph.getTimeRange(this);
+
+    if (this.state.refresh === VariableRefresh.onTimeRangeChanged) {
+      this._subs.add(
+        timeRange.subscribeToState({
+          next: () => {
+            this.updateSubscription = this.validateAndUpdate().subscribe();
+          },
+        })
+      );
+    }
+  }
+
+  public deactivate(): void {
+    super.deactivate();
+    if (this.updateSubscription) {
+      this.updateSubscription.unsubscribe();
+    }
+
+    if (this.dataSourceSubject) {
+      this.dataSourceSubject.unsubscribe();
+    }
+  }
+
+  public getValueOptions(args: VariableGetOptionsArgs): Observable<VariableValueOption[]> {
+    if (this.state.query === '' || !this.state.datasource) {
+      return of([]);
+    }
+
+    return from(getDataSourceSrv().get(this.state.datasource ?? '')).pipe(
+      mergeMap((ds) => {
+        const runner = createQueryVariableRunner(ds);
+        const target = runner.getTarget(this);
+        const request = this.getRequest(target);
+        return runner.runRequest({}, request).pipe(
+          filter((data) => data.state === LoadingState.Done || data.state === LoadingState.Error), // we only care about done or error for now
+          take(1), // take the first result, using first caused a bug where it in some situations throw an uncaught error because of no results had been received yet
+          mergeMap((data: PanelData) => {
+            if (data.state === LoadingState.Error) {
+              return throwError(() => data.error);
+            }
+            return of(data);
+          }),
+          toMetricFindValues(),
+          mergeMap((values) => {
+            let regex = '';
+            if (this.state.regex) {
+              regex = sceneGraph.interpolate(this, this.state.regex, undefined, 'regex');
+            }
+            return of(metricNamesToVariableValues(regex, this.state.sort, values));
+          }),
+          catchError((error) => {
+            if (error.cancelled) {
+              return of([]);
+            }
+            return throwError(() => error);
+          })
+        );
+      })
+    );
+  }
+
+  private getRequest(target: DataQuery) {
+    // TODO: add support for search filter
+    // const { searchFilter } = this.state.searchFilter;
+    // const searchFilterScope = { searchFilter: { text: searchFilter, value: searchFilter } };
+    // const searchFilterAsVars = searchFilter ? searchFilterScope : {};
+    const variableAsVars: ScopedVars = { variable: { text: this.state.text, value: this.state.value } };
+    const scopedVars: ScopedVars = {
+      // ...searchFilterAsVars,
+      ...variableAsVars,
+      __sceneObject: { text: '__sceneObject', value: this },
+    };
```

Comment:
not sure I understand the variableAsVars line, it adds a variable named "variable" to the scopedVars object, why?

Reply:
TBH - I'm not sure and this is just taken from https://github.com/grafana/grafana/blob/scenes-query-variable/public/app/features/variables/query/VariableQueryRunner.ts#L174

Reply:
@dprokop that is very strange, so inside the query definition of a variable you can technically use `${variable.value}` (just tested this and interpolates to the current variable's value). But I don't see this documented or any use case for this. 

It's kind of recursive to include the current value of oneself when evaluating your next value? 

I would remove it, better to get a bug report on it so we can add it and have a unit test and some docs on this behavior. 

I think it's likely just a mistake

Reply:
Agree, removed